### ISSUE TITLE
'Slice tool' acts as 'Rectangle tool' if Tilemap Mode is 'Tiles'

### ISSUE THREAD

**Issue by Gasparoken** [aseprite/aseprite#4306]
### Aseprite and System version

* Aseprite version: 1.3.2
* System: Windows/macOS/Linux

**Comment by Liebranca** [comment]
Besides needing to convert coordinates from tile to canvas (maybe at `SliceInk::inkHLine`?), there's a further issue that arises uppon adding `&& !params.ink->isSlice()` to the check here:

https://github.com/aseprite/aseprite/blob/0f7cac0f0cc93a9a1e06afe5fef3fd641c92eb3b/src/app/ui/editor/tool_loop_impl.cpp#L761-L768

When making a new slice in tiles mode, nothing is drawn to indicate the area being sliced, and at times, a segfault is thrown on my end.

Looking at the coredump, the crash happens at `XorInkProcessing<RgbTraits>::processPixel`, both pointers being dereferenced (`m_dstAddress` and `m_srcAddress`) are invalid from what I can see.

But I can't reproduce this with selection tools, whose ink has the same base class and uses the same ink processing, so I'm assuming the actual problem has to be elsewhere. I'm reading `src/app/tools/intertwine.cpp` and `src/app/tools/point_shape.cpp` to see if I can find anything.

For reference, and just for completeness, here's the backtrace:

```C++
// src/app/tools/ink_processing.h:1086
app::tools::(anonymous namespace)::XorInkProcessing<doc::RgbTraits>::processPixel(this=0x5d9925835930, y=32, x=<optimized out>);

// src/app/tools/ink_processing.h:90
app::tools::(anonymous namespace)::InkProcessing<app::tools::(anonymous namespace)::XorInkProcessing<doc::RgbTraits> >::processScanline(this=0x5d9925835930, x1=48, y=32, x2=<optimized out>, loop=0x5d9924c80390);

// src/app/tools/inks.h:31
app::tools::BaseInk::inkHline(this=0x5d99258c0880, x1=48, y=32, x2=616259552, loop=0x5d9924c80390);

// src/app/tools/inks.h:299
app::tools::SliceInk::inkHline(this=0x5d99258c0880, x1=<optimized out>, y=<optimized out>, x2=<optimized out>, loop=0x5d9924c80390);

// src/app/tools/point_shape.cpp:70
app::tools::PointShape::doInkHline(x1=<optimized out>, y=32, x2=<optimized out>, loop=0x5d9924c80390);

// src/app/tools/point_shapes.h:55
app::tools::TilePointShape::transformPoint(this=<optimized out>, loop=0x5d9924c80390, pt=...);

// src/doc/algo.cpp:124
doc::algo_line_continuous(x0=48, y0=47, x1=62, y1=47, data=0x7fff85423fe0, proc=0x5d99106588a0 <app::tools::Intertwine::doPointshapePointDynamics(int, int, app::tools::Intertwine::LineData*)>);

// src/app/tools/intertwine.cpp:126
app::tools::Intertwine::doPointshapeLine(a=..., b=..., loop=0x5d9924c80390);

// src/app/tools/intertwine.cpp:118
app::tools::Intertwine::doPointshapeLineWithoutDynamics(x1=48, y1=47, x2=62, y2=47, loop=0x5d9924c80390);

// src/app/tools/intertwiners.h:205
app::tools::IntertwineAsRectangles::joinStroke(this=<optimized out>, loop=0x5d9924c80390, stroke=...);

// src/app/tools/tool_loop_manager.cpp:288
app::tools::ToolLoopManager::doLoopStep(this=this@entry=0x5d9924bc12e0, lastStep=false);

// src/app/tools/tool_loop_manager.cpp:209
app::tools::ToolLoopManager::movement(this=0x5d9924bc12e0, pointer=...);

// src/app/ui/editor/drawing_state.cpp:395
app::DrawingState::handleMouseMovement(this=0x5d9925860a30);

// src/app/ui/editor/drawing_state.cpp:295
app::DrawingState::onCommitMouseMove(this=0x5d9925860a30, editor=<optimized out>, spritePos=...);

// src/app/ui/editor/delayed_mouse_move.cpp:75
app::DelayedMouseMove::commitMouseMove(this=0x5d9925860aa0);
```

**Comment by Liebranca** [comment]
For documentation, there were actually *two* possible segfaults introduced by the initial change: the first one was caused by trying to slice outside of the canvas bounds, and the second one was a `nullptr` dereference on `Render::renderImage()` using the `tileFlags` parameter (see: [here](https://github.com/aseprite/aseprite/blob/31c80a5e0c8724635eff173a64b88604de595367/src/render/render.cpp#L757), [here](https://github.com/aseprite/aseprite/blob/31c80a5e0c8724635eff173a64b88604de595367/src/render/render.cpp#L1399) and [here](https://github.com/aseprite/aseprite/blob/31c80a5e0c8724635eff173a64b88604de595367/src/render/render.cpp#L1327)).

Both are seemingly fixed by 7d77545, I'll test it some tomorrow to be sure.
